src/HttpExtractor.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\HttpExtractor;
+
+use function dirname;
+use GuzzleHttp\Client;
+use Psr\Http\Message\UriInterface;
+
+class HttpExtractor
+{
+    /** @var Client */
+    private $client;
+
+    public function __construct(
+        Client $client
+    ) {
+        $this->client = $client;
+    }
+
+    public function extract(UriInterface $httpSource, string $filesystemDestination): void
+    {
+        if (!is_writable(dirname($filesystemDestination))) {
```

Comment:
Na všechny operace s filesystemem se snažím používat `Symfony\Filesystem`, na operace s cestama a souborama `\SplFileInfo`. Mám jistotu, že všechny neplatný operace skončej nějakou správnou exception. Potom je dokonce možná i `!is_writable` zbytečný, protože Guzzle musí stejně vyhodit exception, že tam nejde zapsat. Skript ani end user s tím nic neudělá, takže to skončí jako application error a přistane to v reportingu.

Reply:
jo, jo to iswritable urcite netestovat

Reply:
Opraveno.